database/sqlite/sqlite_aclk_alert.c

```diff
@@ -747,100 +748,95 @@ void aclk_start_alert_streaming(char *node_id, uint64_t batch_id, uint64_t start
-            return;
-        }
+    if (unlikely(!host))
+        return;
 
-        if (unlikely(batch_id == 1) && unlikely(start_seq_id == 1))
-            sql_queue_existing_alerts_to_aclk(host);
-    } else
-        wc = (struct aclk_database_worker_config *)find_inactive_wc_by_node_id(node_id);
+    struct aclk_sync_host_config *wc  = host->aclk_sync_host_config;
 
-    if (likely(wc)) {
-        log_access("ACLK REQ [%s (%s)]: ALERTS STREAM from %"PRIu64" batch=%"PRIu64, node_id, wc->host ? rrdhost_hostname(wc->host) : "N/A", start_seq_id, batch_id);
-        __sync_synchronize();
-        wc->alerts_batch_id = batch_id;
-        wc->alerts_start_seq_id = start_seq_id;
-        wc->alert_updates = 1;
-        wc->pause_alert_updates = 0;
-        __sync_synchronize();
+    if (unlikely(!wc))
+        return;
+
+    if (unlikely(!host->health.health_enabled)) {
+        log_access("ACLK STA [%s (N/A)]: Ignoring request to stream alert state changes, health is disabled.", node_id);
+        return;
     }
-    else
-        log_access("ACLK STA [%s (N/A)]: ACLK synchronization thread is not active.", node_id);
 
-    return;
+    // TODO: CHECK
+    if (unlikely(batch_id == 1) && unlikely(start_seq_id == 1))
+        sql_queue_existing_alerts_to_aclk(host);
+
+    log_access("ACLK REQ [%s (%s)]: ALERTS STREAM from %"PRIu64" batch=%"PRIu64, node_id, wc->host ? rrdhost_hostname(wc->host) : "N/A", start_seq_id, batch_id);
+    wc->alerts_batch_id = batch_id;
+    wc->alerts_start_seq_id = start_seq_id;
+    wc->alert_updates = 1;
 }
 
-void sql_process_queue_removed_alerts_to_aclk(struct aclk_database_worker_config *wc, struct aclk_database_cmd cmd)
+void sql_process_queue_removed_alerts_to_aclk(char *node_id)
 {
-    UNUSED(cmd);
-
-    BUFFER *sql = buffer_create(1024, &netdata_buffers_statistics.buffers_sqlite);
-
-    buffer_sprintf(sql,"insert into aclk_alert_%s (alert_unique_id, date_created, filtered_alert_unique_id) " \
-        "select unique_id alert_unique_id, unixepoch(), unique_id alert_unique_id from health_log_%s " \
-        "where new_status = -2 and updated_by_id = 0 and unique_id not in " \
-        "(select alert_unique_id from aclk_alert_%s) order by unique_id asc " \
-        "on conflict (alert_unique_id) do nothing;", wc->uuid_str, wc->uuid_str, wc->uuid_str);
+    struct aclk_sync_host_config *wc;
+    RRDHOST *host = find_host_by_node_id(node_id);
+    freez(node_id);
 
-    db_execute(buffer_tostring(sql));
+    if (unlikely(!host || !(wc = host->aclk_sync_host_config)))
+        return;
 
-    log_access("ACLK STA [%s (%s)]: QUEUED REMOVED ALERTS", wc->node_id, wc->host ? rrdhost_hostname(wc->host) : "N/A");
+    char sql[ACLK_SYNC_QUERY_SIZE * 2];
 
-    buffer_free(sql);
+    snprintfz(sql,ACLK_SYNC_QUERY_SIZE * 2 - 1, "INSERT INTO aclk_alert_%s (alert_unique_id, date_created, filtered_alert_unique_id) " \
```

Comment:
Here we have another query given directly as argument.

Reply:
We also have more queries inside this file, so if you agree to move all them to `define`, please, take a look in other arguments inside this file.